app/models/user.rb

```diff
@@ -1,6 +1,12 @@
 # frozen_string_literal: true
 
 class User < ApplicationRecord
+  include Discard::Model
+
+  has_many :keywords, inverse_of: :user, dependent: :destroy
```

Comment:
If we are using discard, then its better not to have `dependent: :destroy`, as it would hard delete the keywords. A better approach would be 

```
   class User
     includes Discard:Model

     after_discard do
       keywords.discard_all
    end
end
```

Reply:
I'm not sure to agree 100% ^^
--> As you said, I've implemented the soft-delete on keywords in "anticipation for later need". 
Currently, there is absolutely 0 use case where we want to discard a keyword. 

Assuming we were in the future and had a UI to discard keywords: discarding all keywords when discarding a user is wrong. If the user undiscard his account later, he would not be able to restore the state of discarded keywords (as all would be discarded). Plus the state of the keyword should not depend on the state of the user IMO.

The `dependent` is more here as a convention/best-practice, but because our code does not delete (nor destroy) any user, so it should not be triggered. 

What do you think? :) 

-- 
After a second reading, I think I understood the idea: You were more using `paranoia` as a "delete safety" (because it override delete/destroy), so that no developer would risk to accidentally delete a DB record, right? (in which case `discard` does not make the job as it is adding a method and still provide direct access to delete/destroy). And so... makes sense to setup paranoia by default even without real need for soft-delete ; but it maybe makes less sense to setup discard by default... Not sure if that's the point?

Reply:
@malparty Yes, thats why I recommend using paranoia as we are using it in most of our projects and it is present in our rails template. If you want to change the convention, I guess you can bring it to vote in web chapter. If you use paranoia, we can just add it to all models as safety with just one line.

Reply:
And yea to summarize, it makes better sense to setup paranoia by default on all models, than to setup discard on all models

Reply:
Ok clear! :) Thanks for all the explanations~ 
Well, I do not personally want to change our standards if they work well ;-) 
But it can be interesting indeed to try to understand why paranoia does not recommend itself for new projects. I'll call for a vote, but I'm perfectly fine with paranoia if you and others had no issue with it ;-)

For this IC, if you don't mind I'd be very happy to keep it as-is and to move forward :D